controller/RestController.php

```diff
@@ -683,18 +683,31 @@ public function label($request)
             return $this->returnError(400, "Bad Request", "uri parameter missing");
         }
 
-        $results = $request->getVocab()->getConceptLabel($request->getUri(), $request->getLang());
-        if ($results === null) {
+        $prefResults = $request->getVocab()->getConceptLabel($request->getUri(), $request->getLang());
+        if ($prefResults === null) {
             return $this->returnError('404', 'Not Found', "Could not find concept <{$request->getUri()}>");
         }
+        $otherResults = $request->getVocab()->getOtherConceptLabels($request->getUri(), $request->getLang());
```

Comment:
This code is doing triggering two SPARQL queries (one for the prefLabel, another for altLabel+hiddenLabel). Wouldn't it be better to use just a single SPARQL query that asks for all three label properties at once? Simply adding skos:prefLabel to the VALUES block should be enough. That would be more efficient.

Reply:
Good point! No need to use the old label query in this case at all.